### PR TITLE
ci!: enforce pylint check through Travis

### PR DESCRIPTION
**Pull request by redeboer**
Part 6 of #78 

Required some refactoring because of `too-many-locals`

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -6,13 +6,22 @@
...
 
+[TYPECHECK]
+ignored-modules=
+    amplitf.interface,
+
 [DESIGN]
 # Minimum number of public methods for a class (see R0903). Default is 2.
 # We decrease this to 1, since some interface classes just have a single
 # (public) method
+max-args=7  # default: 5
```
Comment (spflueger) [inline .pylintrc:26]: can you shift the comment down

Comment (redeboer) [reply .pylintrc:26]: Which comment? `# default`?

File: tensorwaves/physics/helicityformalism/amplitude.py [hunk 2]
```diff
@@ -501,56 +504,104 @@ def __call__(self, dataset: dict) -> tf.Tensor:
...
+            recoil_system.recoil_ids = _safe_wrap_list(
+                definition["RecoilSystem"]["RecoilFinalState"]
+            )
+        if "ParentRecoilFinalState" in definition["RecoilSystem"]:
+            recoil_system.parent_recoil_ids = _safe_wrap_list(
+                definition["RecoilSystem"]["ParentRecoilFinalState"]
+            )
+        return recoil_system
+
+
```
Comment (redeboer) [inline tensorwaves/physics/helicityformalism/amplitude.py:561]: So this part goes a bit into #77. It doesn't fix that issue, but it's an attempt at formulating a data structure that can then be filled/created from a recipe file (`dict`).

Anyway, in this PR the motivation for creating these classes was just to reduce the number of local variables under `_create_helicity_decay` that `pylint` was complaining about.

File: tensorwaves/physics/helicityformalism/amplitude.py [hunk 3]
```diff
@@ -501,56 +504,104 @@ def __call__(self, dataset: dict) -> tf.Tensor:
...
+    ) -> None:
+        self.recoil_ids: List[int] = []
+        self.parent_recoil_ids: List[int] = []
+        if recoil_ids:
+            self.recoil_ids = recoil_ids
+        if parent_recoil_ids:
+            self.parent_recoil_ids = parent_recoil_ids
+
+    @staticmethod
+    def from_dict(definition: Dict[str, Any]) -> "_RecoilSystem":
```
Comment (spflueger) [inline tensorwaves/physics/helicityformalism/amplitude.py:549]: I guess those changes where necessary to get rid of the complexity issue from pylint?

Comment (redeboer) [reply tensorwaves/physics/helicityformalism/amplitude.py:549]: https://github.com/ComPWA/tensorwaves/pull/84#discussion_r443465116

File: .pylintrc (hunk 1 above)
Comment (spflueger) [reply .pylintrc:26]: The comment above corresponds to the line below.

File: tensorwaves/physics/helicityformalism/amplitude.py (hunk 2 above)
Comment (spflueger) [reply tensorwaves/physics/helicityformalism/amplitude.py:561]: Oh sorry, so that comment too late. All right.

File: .pylintrc (hunk 1 above)
Comment (redeboer) [reply .pylintrc:26]: Oops yes

**Comment by redeboer** [comment]
Hmm so this shows we need to write more tests...
https://travis-ci.com/github/ComPWA/tensorwaves/jobs/352148558

File: .pylintrc (hunk 1 above)
Comment (redeboer) [reply .pylintrc]: 7bb4b205900ab26f37c27e0ad11ee0eef0cd4ac5

**Pull request merged**